### ISSUE TITLE
lsvmd example app shows incorrect socket ID for backing devices

### ISSUE THREAD

**Issue by tanabarr** [spdk/spdk#2248]
# Bug report

When running the lsvmd example application to list backing devices present behind VMDs, socket ID is always returned as 0 even if the PCI bus ID is mapped to a different socket.

Illustrated by updating lsvmd.c to show socket as well as address and running the recompiled application:
```
tanabarr@wolf-157:~/projects/daos_m/build-leap15/external/dev/spdk> git diff
diff --git a/examples/vmd/lsvmd/lsvmd.c b/examples/vmd/lsvmd/lsvmd.c
index b083bc049..71bc97195 100644
--- a/examples/vmd/lsvmd/lsvmd.c
+++ b/examples/vmd/lsvmd/lsvmd.c
@@ -100,7 +100,7 @@ int main(int argc, char **argv)
                                continue;
                        }

-                       printf("%s\n", addr_buf);
+                       printf("%s, socket %d\n", addr_buf, pci_device->socket_id);
                }
        }

tanabarr@wolf-157:~/projects/daos_m/build-leap15/external/dev/spdk> build/examples/./lsvmd
[2021-11-23 12:18:56.746376] Starting SPDK v21.07 git sha1 adeb04968 / DPDK 21.05.0 initialization...
[2021-11-23 12:18:56.746428] [ DPDK EAL parameters: [2021-11-23 12:18:56.746438] lsvmd [2021-11-23 12:18:56.746446] --no-shconf [2021-11-23 12:18:56.746454] -c 0x1 [2021-11-23 12:18:56.746461] --log-level=lib.eal:6 [2021-11-23 12:18:56.746468] --log-level=lib.cryptodev:5 [2021-11-23 12:18:56.746476] --log-level=user1:6 [2021-11-23 12:18:56.746483] --base-virtaddr=0x200000000000 [2021-11-23 12:18:56.746490] --match-allocations [2021-11-23 12:18:56.746497] --file-prefix=spdk_pid29586 [2021-11-23 12:18:56.746504] ]
EAL: No free 2048 kB hugepages reported on node 1
EAL: No available 1048576 kB hugepages reported
TELEMETRY: No legacy callbacks, legacy socket not created
5d0505:01:00.0, socket 0
5d0505:03:00.0, socket 0
850505:07:00.0, socket 0
850505:09:00.0, socket 0
850505:0b:00.0, socket 0
850505:0d:00.0, socket 0
850505:0f:00.0, socket 0
850505:11:00.0, socket 0
850505:14:00.0, socket 0
d70505:05:00.0, socket 0

tanabarr@wolf-157:~/projects/daos_m/build-leap15/external/dev/spdk> dmesg| grep "NUMA node"
[    1.784232] pci_bus 0000:00: on NUMA node 0
[    1.793327] pci_bus 0000:17: on NUMA node 0
[    1.797829] pci_bus 0000:3a: on NUMA node 0
[    1.800535] pci_bus 0000:5d: on NUMA node 0
[    1.802816] pci_bus 0000:80: on NUMA node 1
[    1.810910] pci_bus 0000:85: on NUMA node 1
[    1.814389] pci_bus 0000:ae: on NUMA node 1
[    1.817159] pci_bus 0000:d7: on NUMA node 1
```

## Expected Behavior

VMD backing device addresses listed through lsvmd beginning with 85 and d7 (signifying PCI bus ID) should show socket ID 1 to match the relevant NUMA node displayed in dmesg.

## Current Behavior

All addresses show socket 0.

## Steps to Reproduce

run the commands listed above

## Context (Environment including OS version, SPDK version, etc.)

SUSE LEAP 15.3, SPDK 21.07

**Comment by jimharris** [comment]
Hi @tanabarr, can you try https://review.spdk.io/gerrit/c/spdk/spdk/+/10373?

**Comment by tanabarr** [comment]
works great thanks Jim:
```
tanabarr@wolf-157:~/projects/spdk> build/examples/lsvmd
[2021-11-24 00:33:03.818335] Starting SPDK v21.07 git sha1 adeb04968 / DPDK 21.05.0 initialization...
[2021-11-24 00:33:03.818390] [ DPDK EAL parameters: [2021-11-24 00:33:03.818400] lsvmd [2021-11-24 00:33:03.818407] --no-shconf [2021-11-24 00:33:03.818414] -c 0x1 [2021-11-24 00:33:03.818420] --log-level=lib.eal:6 [2021-11-24 00:33:03.818426] --log-level=lib.cryptodev:5 [2021-11-24 00:33:03.818433] --log-level=user1:6 [2021-11-24 00:33:03.818449] --base-virtaddr=0x200000000000 [2021-11-24 00:33:03.818455] --match-allocations [2021-11-24 00:33:03.818461] --file-prefix=spdk_pid66237 [2021-11-24 00:33:03.818467] ]
EAL: No free 2048 kB hugepages reported on node 1
EAL: No available 1048576 kB hugepages reported
TELEMETRY: No legacy callbacks, legacy socket not created
5d0505:01:00.0, 0ock 0
5d0505:03:00.0, 0ock 0
850505:07:00.0, 0ock 1
850505:09:00.0, 0ock 1
850505:0b:00.0, 0ock 1
850505:0d:00.0, 0ock 1
850505:0f:00.0, 0ock 1
850505:11:00.0, 0ock 1
850505:14:00.0, 0ock 1
d70505:05:00.0, 0ock 1
``

**Comment by ksztyber** [comment]
Patch was merged, closing.
